internal/outputs/json_collector_test.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package outputs
+
+import (
+	"os"
+	"testing"
+
+	"github.com/google/gke-policy-automation/internal/policy"
+)
+
+type MockFileWriter struct {
+}
+
+func (f MockFileWriter) WriteFile(filename string, data []byte, perm os.FileMode) error {
+	return nil
+}
+
+func TestCollectingToJson(t *testing.T) {
+
+	evalResults := make([]*policy.PolicyEvaluationResult, 0)
+	r := policy.NewPolicyEvaluationResult()
+
+	policyTitle := "title"
+	policyDescription := "description"
+	policyGroup := "group"
+	policyValid := true
+
+	r.AddPolicy(&policy.Policy{
+		Title:            policyTitle,
+		Description:      policyDescription,
+		Group:            policyGroup,
+		Valid:            policyValid,
+		Violations:       []string{},
+		ProcessingErrors: []error{},
+	})
+
+	evalResults = append(evalResults, r)
+
+	var collector = NewJSONResultToCustomWriterCollector("sample.json", MockFileWriter{})
```

Comment:
Good one with a mock!
I would extend it and also check if the data looks fine from the mock perspective.
You can use the trick below to define mock behaviour in a test.

Define a mock in a such way:
```
type myMock struct {
  WriteFileFn func(filename string, data []byte, perm os.FileMode) error
}

func (m myMock) WriteFile(filename string, data []byte, perm os.FileMode) error {
  return m.WriteFileFn(filename, data, perm)
}
```

Now you can do that in a test:
```
func TestMyMock(t *testing.T) {
  //given
  outFileName := "output.json"
  mock := myMock{}
  mock.WriteFileFn = func(filename string, data []byte, perm os.FileMode) {
    if filename != outFileName {
    // t.errorf mock got different file name than anticipated
    }
  }
  //when
  var collector = NewJSONResultToCustomWriterCollector(outFileName, myMock)
  ...
}
```

Reply:
will be added in next PR (with config options added)